src/EnergyPlus/ExtendedHI.cc

```diff
@@ -0,0 +1,538 @@
+    constexpr Real64 Za_bar = 60.6 / 11.6; // Pa m^2/W, mass transfer resistance through air, clothed part of skin
+    constexpr Real64 Za_un = 60.6 / 12.3;  // Pa m^2/W, mass transfer resistance through air, when being naked
+
+    constexpr Real64 tol = 1e-8;
+    constexpr Real64 maxIter = 100;
+    Real64 find_eqvar_phi(EnergyPlusData &state, Real64 Ta, Real64 RH)
+    {
+
+        Real64 phi = 0.84;
+        Real64 Pa = RH * pvstar(Ta);
+        Real64 Rs = 0.0387;
+        Real64 m = (Pc - Pa) / (Zs(Rs) + Za);
+        Real64 Ts;
+        int SolFla;
+        General::SolveRoot(
+            state,
+            tol,
+            maxIter,
+            SolFla,
+            Ts,
+            [&](Real64 Ts) { return (Ts - Ta) / Ra(Ts, Ta) + (Pc - Pa) / (Zs(Rs) + Za) - (Tc - Ts) / Rs; },
+            std::max(0.0, std::min(Tc, Ta) - Rs * std::abs(m)),
+            std::max(Tc, Ta) + Rs * std::abs(m));
+        Real64 flux1 = Q - Qv(Ta, Pa) - (1.0 - phi) * (Tc - Ts) / Rs;
+        if (flux1 <= 0.0) {
+            phi = 1.0 - (Q - Qv(Ta, Pa)) * Rs / (Tc - Ts);
+        }
+        return phi;
+    }
+
+    Real64 find_eqvar_Rf(EnergyPlusData &state, Real64 Ta, Real64 RH)
+    {
+        Real64 Pa = RH * pvstar(Ta);
+        Real64 Rs = 0.0387;
+        Real64 phi = 0.84;
+        Real64 m_bar = (Pc - Pa) / (Zs(Rs) + Za_bar);
+        Real64 m = (Pc - Pa) / (Zs(Rs) + Za);
+        Real64 Ts;
+        int SolFla;
+        General::SolveRoot(
+            state,
+            tol,
+            maxIter,
+            SolFla,
+            Ts,
+            [&](Real64 Ts) { return (Ts - Ta) / Ra(Ts, Ta) + (Pc - Pa) / (Zs(Rs) + Za) - (Tc - Ts) / Rs; },
+            std::max(0.0, std::min(Tc, Ta) - Rs * std::abs(m)),
+            std::max(Tc, Ta) + Rs * std::abs(m));
+        Real64 Tf;
+        General::SolveRoot(
+            state,
+            tol,
+            maxIter,
+            SolFla,
+            Tf,
+            [&](Real64 Tf) { return (Tf - Ta) / Ra_bar(Tf, Ta) + (Pc - Pa) / (Zs(Rs) + Za_bar) - (Tc - Tf) / Rs; },
+            std::max(0.0, std::min(Tc, Ta) - Rs * std::abs(m_bar)),
+            std::max(Tc, Ta) + Rs * std::abs(m_bar));
+        Real64 flux1 = Q - Qv(Ta, Pa) - (1.0 - phi) * (Tc - Ts) / Rs;
+        Real64 flux2 = Q - Qv(Ta, Pa) - (1.0 - phi) * (Tc - Ts) / Rs - phi * (Tc - Tf) / Rs;
+        Real64 Rf;
+        if (flux1 <= 0.0) {
+            Rf = std::numeric_limits<Real64>::infinity();
+        } else if (flux2 <= 0.0) {
+            Real64 Ts_bar = Tc - (Q - Qv(Ta, Pa)) * Rs / phi + (1.0 / phi - 1.0) * (Tc - Ts);
+            General::SolveRoot(
+                state,
+                tol,
+                maxIter,
+                SolFla,
+                Tf,
+                [&](Real64 Tf) {
+                    return (Tf - Ta) / Ra_bar(Tf, Ta) + (Pc - Pa) * (Tf - Ta) / ((Zs(Rs) + Za_bar) * (Tf - Ta) + r * Ra_bar(Tf, Ta) * (Ts_bar - Tf)) -
+                           (Tc - Ts_bar) / Rs;
```

Comment:
This is definitely getting pretty long to just be a lambda argument.  I'd consider making this a nicely named function with clear arguments, etc., and then just have the lambda call that function.  Debugging an issue with the arithmetic in this line would be quite difficult.

Reply:
Yeah, indeed. This is probably better separated out.